File: src/main/java/io/zeitwert/ddd/oe/model/ObjUser.java

package io.zeitwert.ddd.oe.model;

import io.zeitwert.ddd.obj.model.Obj;

public interface ObjUser extends Obj {

	String getEmail();

	void setEmail(String email);

	String getName();

	void setName(String name);

	String getDescription();

	void setDescription(String description);

}

File: fm-domain/src/main/java/io/zeitwert/fm/oe/model/ObjUser.kt
package io.zeitwert.fm.oe.model

import dddrive.app.obj.model.Obj
import dddrive.ddd.property.model.ReferenceSetProperty
import io.zeitwert.fm.dms.model.ObjDocument
import io.zeitwert.fm.oe.model.enums.CodeUserRole

interface ObjUser : Obj {

	var email: String?

	var password: String?

	var needPasswordChange: Boolean?

	var name: String?

	var description: String?

	val isAppAdmin: Boolean

	val isAdmin: Boolean

	var avatarImageId: Any?

	val avatarImage: ObjDocument?

	var role: CodeUserRole?

	fun hasRole(role: CodeUserRole): Boolean

	val tenantSet: ReferenceSetProperty<ObjTenant>

}
